src/bellmanford.jl

```diff
@@ -105,3 +108,28 @@ function has_negative_edge_cycle{V, D}(
     edge_inspector = VectorEdgePropertyInspector{D}(edge_dists)
     has_negative_edge_cycle(graph, edge_inspector)
 end
+
+function enumerate_paths{V,D<:Number}(state::BellmanFordStates{V,D}, dest::Vector{V})
+    parents = state.parents
+    hasparent = state.hasparent
+    
+    num_dest = length(dest)
+    all_paths = Array(Vector{V},num_dest)
+    for i=1:num_dest
+        all_paths[i] = V[]
+        index = dest[i]
+        if hasparent[index] || parents[index] == index
+            while hasparent[index]
```

Comment:
I haven't actually thought about the case for listing the paths to all possible locations. The `parents` vector is always available (if you're talking about memoizing intermediate paths), hence my understanding that `enumerate_paths` is merely a convenience function.

Anyhow, `enumerate_paths` was meant as a replacement for `dijkstra_shortest_paths_explicit`, which does the same thing, and I'm not familiar with performance optimizations for it. My understanding is that in huge graphs, you seldom ever actually want to store the paths to all vertices (in memory) anyway, esp. when you already have `parents`.